plumbing/format/packfile/decoder.go

```diff
@@ -128,9 +138,20 @@ func (d *Decoder) Decode() (checksum plumbing.Hash, err error) {
 	return d.s.Checksum()
 }
 
-func (d *Decoder) doDecode() error {
+func (d *Decoder) Count() (uint32, error) {
 	_, count, err := d.s.Header()
 	if err != nil {
+		return 0, err
+	}
+
+	d.cache.SetMaxElements(int(count / magicCacheNumber))
```

Comment:
Setting the max elements of cache inside this method (public Count) looks wierd.